File: Examples/CSharp/ProgrammersGuide/Working-with-Pages/CopyVisioPage.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSharp.ProgrammersGuide.Working_with_Pages
{
    public class CopyVisioPage
    {
        public static void Run()
        {
            //ExStart:CopyVisioPage
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_VisioPages();

            // initialize the new visio diagram
            Diagram NewDigram = new Diagram();

            // load source diagram
            Diagram dgm = new Diagram(dataDir + "Drawing1.vsdx");
            // add all masters from the source Visio diagram
            foreach (Master master in dgm.Masters)
                NewDigram.Masters.Add(master);

            // get page object
            Aspose.Diagram.Page SrcPage = dgm.Pages.GetPage("Page-1");
            // Set name
            SrcPage.Name = "new page";

            // it calculates max page id
            int max = 0;
            if (NewDigram.Pages.Count != 0)
                max = NewDigram.Pages[0].ID;

            for (int i = 1; i < NewDigram.Pages.Count; i++)
            {
                if (max < NewDigram.Pages[i].ID)
                    max = NewDigram.Pages[i].ID;
            }
            
            // set max page ID 
            int MaxPageId = max;
            // Set page ID
            SrcPage.ID = MaxPageId + 1;

            // add page from the source diagram
            NewDigram.Pages.Add(SrcPage);
            // remove first empty page
            NewDigram.Pages.Remove(NewDigram.Pages[0]);
            // save diagram
            NewDigram.Save(dataDir + "CopyVisioPage_Out.vsdx", SaveFileFormat.VSDX);
            //ExEnd:CopyVisioPage
        }
    }
}

File: Examples/VisualBasic/Working-with-Pages/CopyVisioPage.vb
Imports Aspose.Diagram
Imports System
Public Class CopyVisioPage
    Public Shared Sub Run()
        ' ExStart:CopyVisioPage
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_VisioPages()

        ' Initialize the new visio diagram
        Dim NewDigram As New Diagram()

        ' Load source diagram
        Dim dgm As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))
        ' Add all masters from the source Visio diagram
        For Each master As Master In dgm.Masters
            NewDigram.Masters.Add(master)
        Next

        ' Get page object
        Dim SrcPage As Aspose.Diagram.Page = dgm.Pages.GetPage("Page-1")
        ' Set name
        SrcPage.Name = "new page"

        ' It calculates max page id
        Dim max As Integer = 0
        If NewDigram.Pages.Count <> 0 Then
            max = NewDigram.Pages(0).ID
        End If

        For i As Integer = 1 To NewDigram.Pages.Count - 1
            If max < NewDigram.Pages(i).ID Then
                max = NewDigram.Pages(i).ID
            End If
        Next

        ' Set max page ID 
        Dim MaxPageId As Integer = max
        ' Set page ID
        SrcPage.ID = MaxPageId + 1

        ' Add page from the source diagram
        NewDigram.Pages.Add(SrcPage)
        ' Remove first empty page
        NewDigram.Pages.Remove(NewDigram.Pages(0))
        ' Save diagram
        NewDigram.Save(dataDir & Convert.ToString("CopyVisioPage_out.vsdx"), SaveFileFormat.VSDX)
        ' ExEnd:CopyVisioPage
    End Sub
End Class
